.github/workflows/qa.yaml

```diff
@@ -0,0 +1,62 @@
+name: QA
+# QA checks on a regular Github runner, allowing for concurrent runs.
+# The tests run on a mock back-end.
+
+on:
+  pull_request:
+  workflow_dispatch:
+  push:
+    branches: [main]
+
+jobs:
+  quality:
+    name: "Quality checks"
+    # This job ensures the project compiles and the linter passes before commiting
+    # to the whole Github-Azure-WSL orchestra.
+    strategy:
+      matrix:
+        os: [ubuntu, windows]
+    runs-on: ${{ matrix.os }}-latest
+    steps:
+    - name: Set up Git
+      shell: bash
+      run : |
+        git config --global core.autocrlf false
+        git config --global core.eol lf
+    - name: Check out repository
+      uses: actions/checkout@v3
+    - name: Set up Go
+      uses: actions/setup-go@v3
+      with:
+        go-version-file: go.mod
+    - name: Build
+      shell: bash
+      run: |
+        go build ./...
+    - name: Lint with mock back-end
+      uses: golangci/golangci-lint-action@v3
+      with:
+        version: v1.52.0
+        args:  --build-tags="gowslmock"
+    - name: Lint with real back-end
+      uses: golangci/golangci-lint-action@v3
+      with:
+        version: v1.52.0
+    - name: Prepare repo
+      shell: bash
+      run: |
+        # Prepare repo with mock rootfs and run mock tests
+        mkdir -p images
+        touch images/empty.tar.gz
```

Comment:
I don’t remember why we need those for the mock? It seems that we shouldn’t to me, as we are not going to use the real WSL layer. At worse, the mock can create them when being instantiated and pointing tests to it, wdyt?

Reply:
If the rootfs does not exist, the wslApi syscall fails without explaining why, so I added a check before calling the back-end to make the error less obscure (similar to checking if the distro exists before registering). The point is, this check is in the front-end of GoWSL.

I could either move it to the back-end, or have the mock tests create the file. I think the first option makes more sense, but I'd do it in a diferent PR as this goes beyond enhancing the CI.

Reply:
Do you mind opening a bug report for this? I agree, let’s proceed with this one.

Reply:
Nevermind, I think putting it in TestMain is the best option, this way `TestRegister` requires no modification. I'll do it in a follow-up PR.